Abhi-AnimeX/app/src/main/java/com/abhijith/animex/ui/screens/animelist/AnimeList.kt

```diff
@@ -6,25 +6,20 @@ import androidx.compose.runtime.Composable
 import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
-import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.navigation.NavController
-import com.abhijith.animex.domain.model.AnimeItem
-import com.abhijith.animex.domain.usecases.GetAnimeListUseCase
+import com.abhijith.animex.domain.models.AnimeItem
 import com.abhijith.animex.ui.navigation.Screen
 import com.abhijith.animex.ui.screens.animelist.viewmodel.AnimeListViewModel
-import com.abhijith.animex.ui.screens.animelist.viewmodel.AnimeListViewModelFactory
 import com.abhijith.animex.ui.screens.error.ErrorScreen
 import com.abhijith.animex.ui.screens.loading.LoadingScreen
 import com.google.gson.Gson
 
 @Composable
 fun AnimeList(navController: NavController) {
 
-    // TODO (issue 11)
-    val getAnimeListUseCase = GetAnimeListUseCase()
-    val factory = AnimeListViewModelFactory(getAnimeListUseCase)
-    val viewModel: AnimeListViewModel = viewModel(factory = factory)
-
+    val viewModel: AnimeListViewModel = hiltViewModel()
```

Comment:
about about putting this as a function param above?

Reply:
moved to being a function param